src/main/java/mekanism/client/render/item/ChemicalFluidBarDecorator.java

```diff
@@ -0,0 +1,116 @@
+        this.chemicalCaps = chemicalCaps;
+        this.visibleFor = visibleFor;
+    }
+
+    @Override
+    public boolean render(Font font, ItemStack stack, int xOffset, int yOffset, float blitOffset) {
+        if (!visibleFor.test(stack))
+            return false;
+        RenderSystem.disableDepthTest();
+        RenderSystem.disableTexture();
+        RenderSystem.disableBlend();
+        Tesselator tesselator = Tesselator.getInstance();
+        BufferBuilder bufferBuilder = tesselator.getBuilder();
+        yOffset = yOffset + 12;
+        for (Capability<? extends IChemicalHandler<?,?>> chemicalCap : chemicalCaps) {
+            Optional<? extends IChemicalHandler<?, ?>> capabilityInstance = stack.getCapability(chemicalCap).resolve();
+            if (capabilityInstance.isPresent()) {
+                IChemicalHandler<?, ?> chemicalHandler = capabilityInstance.get();
+                if (chemicalHandler.getTanks() == 0) {
+                    continue;
+                }
+                int tank = 0;
+                if (chemicalHandler.getTanks() > 1 && Minecraft.getInstance().level != null) {
+                    //Cycle through multiple tanks every second, to save some space if multiple tanks are present
+                    tank = (int) (Minecraft.getInstance().level.getGameTime() / 20) % chemicalHandler.getTanks();
+                }
+                renderBar(bufferBuilder, xOffset, yOffset, chemicalHandler.getChemicalInTank(tank), chemicalHandler.getTankCapacity(tank));
+                yOffset--;
+            }
+        }
+
+        if (showFluid) {
+            Optional<IFluidHandlerItem> capabilityInstance = FluidUtil.getFluidHandler(stack).resolve();
+            if (capabilityInstance.isPresent()) {
+                IFluidHandlerItem fluidHandler = capabilityInstance.get();
+                if (fluidHandler.getTanks() != 0) {
+                    int tank = 0;
+                    if (fluidHandler.getTanks() > 1 && Minecraft.getInstance().level != null) {
+                        //Cycle through multiple tanks every second, to save some space if multiple tanks are present
+                        tank = (int) (Minecraft.getInstance().level.getGameTime() / 20) % fluidHandler.getTanks();
+                    }
+                    renderBar(bufferBuilder, xOffset, yOffset, fluidHandler.getFluidInTank(tank), fluidHandler.getTankCapacity(tank));
+                }
+            }
+        }
+        return true;
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, ChemicalStack<?> stack, long capacity) {
+        int color = stack.getChemicalColorRepresentation();
+        renderBar(bufferBuilder, stackXPos, yPos, (float)((double) stack.getAmount() / capacity), color);
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, FluidStack stack, long capacity) {
+        int color = IClientFluidTypeExtensions.of(stack.getFluid()).getTintColor(stack);
```

Comment:
Just split `FluidUtil#getRGBDurabilityForDisplay` into two methods one that accepts an `ItemStack` like it currently does (and grabs the fluid stack), and the other that takes the fluid stack directly. That way we can keep our special case handling of lava and stuff in case we need it in the future.